glib2/ext/glib2/rbglib_regex.c

```diff
@@ -76,6 +76,52 @@ rg_match_flags(VALUE self)
     return UINT2NUM(g_regex_get_match_flags(_SELF(self)));
 }
 
+static VALUE
+rg_split(gint argc, VALUE *argv, VALUE self)
+{
+    VALUE rb_string, rb_string_len, rb_start_position,
+          rb_match_options, rb_max_tokens, rb_options;
+    GError *error = NULL;
+    gchar **strings;
+    const gchar *string;
+    gssize string_len = -1;
+    gint start_position = 0;
+    GRegexMatchFlags match_options = 0;
+    gint max_tokens = 0;
+
+    rb_scan_args(argc, argv, "11", &rb_string, &rb_options);
+
+    rbg_scan_options(rb_options,
+                     "string_len", &rb_string_len,
```

Comment:
We should use `RSTRING_LEN(rb_string)` for `string_len` instead of specifying as an option.